scapy/layers/can.py

```diff
@@ -180,6 +180,38 @@ class CANFD(CAN):
         StrLenField('data', b'', length_from=lambda pkt: int(pkt.length)),
     ]
 
+    def post_build(self, pkt, pay):
+        # type: (bytes, bytes) -> bytes
+
+        data = super(CANFD, self).post_build(pkt, pay)
+
+        length = orb(data[4])
+
+        if 8 < length <= 12:
+            wire_length = 12
+        elif 12 < length <= 16:
+            wire_length = 16
+        elif 16 < length <= 20:
+            wire_length = 20
+        elif 20 < length <= 24:
+            wire_length = 24
+        elif 24 < length <= 32:
+            wire_length = 32
+        elif 32 < length <= 40:
+            wire_length = 40
+        elif 40 < length <= 48:
+            wire_length = 48
+        elif 48 < length <= 56:
+            wire_length = 56
+        elif 56 < length <= 64:
+            wire_length = 64
+        else:
+            wire_length = length
+
```

Comment:
Sometimes padding is different than 0x00. Do you see having it configurable (byte value used as padding, as well as option to remove it entirely) for both CAN and CAN FD as valuable?